### PR TITLE
(Trivial) (DragonFly) Some of the core/stdc/math c-library functions had the 'pure' attribute set

### PR DESCRIPTION
**Pull request by dkgroot**
Phobos Compile Error:
std/math.d(4598): Error: `pure` function `std.math.nearbyint` cannot call impure function `core.stdc.math.nearbyintl`

Issue got apparent when https://github.com/dlang/phobos/pull/6409 was merged.
Note: This fix only references DragonFly specific code

### CODE REVIEW

**Comment by dkgroot** [comment]
I do have one little question, why aren't most/all of the extern c-library math function calls 'pure'? I do not see how any of them could have side-effects. The pure attribute seems to be scattered around a little here. Maybe i am missing something :-)

**Comment by n8sh** [comment]
@dkgroot The source for which functions are annotated `pure` is @ibuclaw's comment https://github.com/dlang/druntime/pull/2045#issuecomment-361695173:
>I'll use [gcc](https://github.com/gcc-mirror/gcc/blob/master/gcc/builtins.def) as a reference, math functions that set errno are marked as MATHFN_ERRNO

File: src/core/stdc/math.d [hunk 1]
```diff
@@ -3378,25 +3378,25 @@ else version(DragonFlyBSD)
     double cbrt(double x);
```
Comment (n8sh) [inline src/core/stdc/math.d]: `cbrt` not pure?

File: src/core/stdc/math.d [hunk 2]
```diff
@@ -3500,81 +3500,81 @@ else version(DragonFlyBSD)
...
-    real nexttowardl(real x, real y);
+    pure real copysignl(real x, real y);
+    pure real nanl(const char *);
+    pure real nextafterl(real x, real y);
+    pure real nexttowardl(real x, real y);
 
-    real fdiml(real x, real y);
-    real fmaxl(real x, real y);
-    real fminl(real x, real y);
+    pure real fdiml(real x, real y);
```
Comment (n8sh) [inline src/core/stdc/math.d]: For non-DragonFlyBSD `fdiml` isn't marked pure. Same for `copysign`, `nan`, `nexttoward`, `nextafter`.

File: src/core/stdc/math.d [hunk 3]
```diff
@@ -3500,81 +3500,81 @@ else version(DragonFlyBSD)
...
+    pure real cbrtl(real x);
+    pure real fabsl(real x);
+    pure real hypotl(real x, real y);
+    pure real powl(real x, real y);
+    pure real sqrtl(real x);
+
+    pure real erfl(real x);
+    pure real erfcl(real x);
+    pure real lgammal(real x);
+    pure real tgammal(real x);
```
Comment (n8sh) [inline src/core/stdc/math.d]: For non-DragonFlyBSD `erf` is marked pure but not `erfc`, `lgamma`, `tgamma`.

File: src/core/stdc/math.d [hunk 4]
```diff
@@ -3500,81 +3500,81 @@ else version(DragonFlyBSD)
...
+    pure real logbl(real x);
+    pure real modfl(real x, real *iptr);
+    pure real scalbnl(real x, int y);
+    pure real scalblnl(real x, c_long y);
+
+    pure real cbrtl(real x);
+    pure real fabsl(real x);
+    pure real hypotl(real x, real y);
+    pure real powl(real x, real y);
+    pure real sqrtl(real x);
```
Comment (n8sh) [inline src/core/stdc/math.d]: For non-DragonFlyBSD `cbrt` and `fabs` are marked pure but not `hypot`, `pow`, `sqrt`.

File: src/core/stdc/math.d [hunk 5]
```diff
@@ -3500,81 +3500,81 @@ else version(DragonFlyBSD)
...
-    real tgammal(real x);
-
-    real ceill(real x);
-    real floorl(real x);
-    real nearbyintl(real x);
-    real rintl(real x);
+    pure real logbl(real x);
+    pure real modfl(real x, real *iptr);
+    pure real scalbnl(real x, int y);
+    pure real scalblnl(real x, c_long y);
```
Comment (n8sh) [inline src/core/stdc/math.d]: For non-DragonFlyBSD `modf` is marked pure but not `logb`, `scalb`.

File: src/core/stdc/math.d [hunk 6]
```diff
@@ -3500,81 +3500,81 @@ else version(DragonFlyBSD)
...
+    pure real sinhl(real x);
+    pure real tanhl(real x);
+
+    pure real expl(real x);
+    pure real exp2l(real x);
+    pure real expm1l(real x);
+    pure real frexpl(real x, int *exp);
     int ilogbl(real x);
-    real ldexpl(real x, int exp);
+    pure real ldexpl(real x, int exp);
```
Comment (n8sh) [inline src/core/stdc/math.d]: For non-DragonFlyBSD `frexp` is marked pure but not `exp`, `exp2`, `expm1`, `ldexp`.

File: src/core/stdc/math.d [hunk 7]
```diff
@@ -3500,81 +3500,81 @@ else version(DragonFlyBSD)
...
+    pure real cosl(real x);
+    pure real sinl(real x);
+    pure real tanl(real x);
+
+    pure real acoshl(real x);
+    pure real asinhl(real x);
+    pure real atanhl(real x);
+    pure real coshl(real x);
+    pure real sinhl(real x);
+    pure real tanhl(real x);
```
Comment (n8sh) [inline src/core/stdc/math.d]: For non-DragonFlyBSD `tanh` and `asinh` are marked `pure` but not `acosh`, `atanh`, `cosh`, `sinh`.

File: src/core/stdc/math.d [hunk 8]
```diff
@@ -3500,81 +3500,81 @@ else version(DragonFlyBSD)
...
-    real tanhl(real x);
-
-    real expl(real x);
-    real exp2l(real x);
-    real expm1l(real x);
-    real frexpl(real x, int *exp);
+    pure real acosl(real x);
+    pure real asinl(real x);
+    pure real atanl(real x);
+    pure real atan2l(real x, real y);
```
Comment (n8sh) [inline src/core/stdc/math.d]: For non-DragonFlyBSD `atan2` is not marked `pure`

Comment (dkgroot) [reply src/core/stdc/math.d]: Thanks for checking all of these. I guess i thought i saw a pattern when making the corrections, which was faulty.

File: src/core/stdc/math.d (hunk 1 above)
Comment (dkgroot) [reply src/core/stdc/math.d]: Added pure to the cbrt function... Thanks for checking !

**Comment by dkgroot** [comment]
@n8sh Thanks got checking ! Much appreciated.

**Comment by ibuclaw** [comment]
Check whitespace is failing.

**Pull request merged**